common/unit_test/Test_Padding.cpp

```diff
@@ -43,434 +65,535 @@ void test_reshape1D_1DView(bool is_C2R) {
   }
 }
 
-void test_reshape2D_3DView(bool is_C2R) {
+template <typename T>
+void test_reshape2D_3DView() {
   const int n0 = 30, n1 = 15, n2 = 8;
-  View3D<double> x("x", n0, n1, n2);
+  bool is_C2R = !KokkosFFT::Impl::is_complex<T>::value;
+  View3D<T> x_out("x_out", n0, n1, n2);
   constexpr int DIM = 3;
   shape_type<3> default_shape({n0, n1, n2});
 
   for (int axis0 = 0; axis0 < DIM; axis0++) {
     for (int axis1 = 0; axis1 < DIM; axis1++) {
       if (axis0 == axis1) continue;
       axes_type<2> axes({axis0, axis1});
+      shape_type<3> in_shape = default_shape;
+      in_shape.at(axis1)     = get_c2r_shape(x_out.extent(axis1), is_C2R);
+      auto [_n0, _n1, _n2]   = in_shape;
+      View3D<Kokkos::complex<double>> x_in("x_in", _n0, _n1, _n2);
       for (int i0 = -1; i0 <= 1; i0++) {
         for (int i1 = -1; i1 <= 1; i1++) {
           shape_type<3> ref_shape = default_shape;
-          std::size_t n0_new  = static_cast<std::size_t>(x.extent(axis0) + i0);
-          std::size_t n1_new  = static_cast<std::size_t>(x.extent(axis1) + i1);
+          std::size_t n0_new =
+              static_cast<std::size_t>(x_in.extent(axis0) + i0);
+          std::size_t n1_new =
+              static_cast<std::size_t>(x_in.extent(axis1) + i1);
           ref_shape.at(axis0) = n0_new;
           ref_shape.at(axis1) = get_c2r_shape(n1_new, is_C2R);
 
           shape_type<2> new_shape = {n0_new, n1_new};
 
           auto modified_shape =
-              KokkosFFT::Impl::get_modified_shape(x, new_shape, axes, is_C2R);
+              KokkosFFT::Impl::get_modified_shape(x_in, x_out, new_shape, axes);
           EXPECT_TRUE(modified_shape == ref_shape);
         }
       }
     }
   }
 }
 
-void test_reshape2D_4DView(bool is_C2R) {
+template <typename T>
+void test_reshape2D_4DView() {
   const int n0 = 5, n1 = 11, n2 = 10, n3 = 8;
-  View4D<double> x("x", n0, n1, n2, n3);
+  bool is_C2R = !KokkosFFT::Impl::is_complex<T>::value;
+  View4D<T> x_out("x_out", n0, n1, n2, n3);
   constexpr int DIM = 4;
   shape_type<4> default_shape({n0, n1, n2, n3});
 
   for (int axis0 = 0; axis0 < DIM; axis0++) {
     for (int axis1 = 0; axis1 < DIM; axis1++) {
       if (axis0 == axis1) continue;
       axes_type<2> axes({axis0, axis1});
+      shape_type<4> in_shape    = default_shape;
+      in_shape.at(axis1)        = get_c2r_shape(x_out.extent(axis1), is_C2R);
+      auto [_n0, _n1, _n2, _n3] = in_shape;
+      View4D<Kokkos::complex<double>> x_in("x_in", _n0, _n1, _n2, _n3);
       for (int i0 = -1; i0 <= 1; i0++) {
         for (int i1 = -1; i1 <= 1; i1++) {
           shape_type<4> ref_shape = default_shape;
-          std::size_t n0_new  = static_cast<std::size_t>(x.extent(axis0) + i0);
-          std::size_t n1_new  = static_cast<std::size_t>(x.extent(axis1) + i1);
+          std::size_t n0_new =
+              static_cast<std::size_t>(x_in.extent(axis0) + i0);
+          std::size_t n1_new =
+              static_cast<std::size_t>(x_in.extent(axis1) + i1);
```

Comment:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.